packages/idempotency/src/idempotentDecorator.ts

```diff
@@ -32,6 +32,7 @@ const idempotent = function (
       const functionPayloadtoBeHashed = isFunctionOption(options)
         ? record[(options as IdempotencyFunctionOptions).dataKeywordArgument]
         : record;
+
```

Comment:
Even with the current signature (& the comment about future refactoring), can we not do any assertion when `idempotentLambdaHandler` is used on the second argument like we do for the fn wrapper?

Like, getting the second argument which might or might not be context ([example](https://github.com/am29d/aws-lambda-powertools-typescript/blob/a35c8ed7154aa834fd50f61dc171b012f0af4874/packages/idempotency/src/makeFunctionIdempotent.ts#L61)), and then doing an assertion on its content ([example](https://github.com/am29d/aws-lambda-powertools-typescript/blob/a35c8ed7154aa834fd50f61dc171b012f0af4874/packages/idempotency/src/makeFunctionIdempotent.ts#L68-L71)).

Is it worth it? Should we not do it?

Reply:
Good point, adding args to check the context works. I have added a copy of `isContext`, it needs a better place. Exporting would mix wrapper and decorator imports. What do you think?

Reply:
It's fine for now, we should probably move it to `commons` together with the other type guards I'm adding in #1528.

Let's leave this for now.